src/wallet.cpp

```diff
@@ -3002,6 +3002,7 @@ bool CWallet::CreateCoinStake(
 
     if (listInputs.empty()) {
         LogPrintf("CreateCoinStake(): listInputs empty\n");
+        MilliSleep(50000);
```

Comment:
It would be better to get the utxo list and calculate the sleep time properly (loop over the available utxo and check how much time left to have the next stakable utxo) instead of hardcode the 50 seconds sleep time.

If there are no outputs then wait at least the min stakable maturity.

Reply:
this particular line addressed an outlier situation wherein the wallet had no stakable inputs and was smashing the stake thread process with repeated locks; ie, there were **NO** available UTXOs to speak of.

for simplicity's sake, i'd much rather prefer this static solution rather than an attempt at calculating the possibility of future valid UTXOs.

Reply:
the problem of any static hardcoded time is that it will try to stake, getting locks and consuming processor time when it has no stakable outputs (same as was doing before but 50 sec spaced). When the calculation of future stakes is actually pretty straightforward, mint height for zpiv and min age for regular piv.

Aside from that, 50 seconds sleep time it's better than before for sure. Not a requirement, just an improvement.

Reply:
yeah, this arbitrary sleep time was put in to specifically combat a lock bashing when no stakable inputs were found. prior to this sleep time, the lock bashing resulted in the client not being able to process/accept new blocks. I felt that an arbitrary 50s sleep here was sufficient enough to negate such behavior without further major rewrite to the code base.